From 89ef77d126d342a13b3ddd6c90130805e2c3e2e9 Mon Sep 17 00:00:00 2001
From: Francisco Molina <femolina@uc.cl>
Date: Thu, 22 Apr 2021 11:46:49 +0200
Subject: [PATCH 4/6] include/nanocbor: add RFC7049 tags

---
 include/nanocbor/nanocbor.h | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/include/nanocbor/nanocbor.h b/include/nanocbor/nanocbor.h
index 016d2c5..b9a87f7 100644
--- a/include/nanocbor/nanocbor.h
+++ b/include/nanocbor/nanocbor.h
@@ -91,6 +91,18 @@ extern "C" {
 #define NANOCBOR_SIZE_INDEFINITE    31U /**< Indefinite sized container */
 /** @} */
 
+/**
+ * @name CBOR Tag values
+ * @{
+ */
+#define NANOCBOR_TAG_DATE_TIME      (0x0) /**< Standard date/time string */
+#define NANOCBOR_TAG_EPOCH          (0x1) /**< Epoch-based date/time */
+#define NANOCBOR_TAG_BIGNUMS_P      (0x2) /**< Positive bignum */
+#define NANOCBOR_TAG_BIGNUMS_N      (0x3) /**< Negative bignum */
+#define NANOCBOR_TAG_DEC_FRAC       (0x4) /**< Decimal Fraction */
+#define NANOCBOR_TAG_BIGFLOATS      (0x5) /**< Bigfloat */
+/** @} */
+
 /**
  * @brief NanoCBOR decoder errors
  */
-- 
2.25.1

