From 1653380dca0acea5dfacebfe26227b84aae3bfe6 Mon Sep 17 00:00:00 2001
From: Benjamin Valentin <benjamin.valentin@ml-pa.com>
Date: Mon, 6 Nov 2023 17:22:05 +0100
Subject: [PATCH] make struct ext4_mountpoint public

---
 include/ext4.h | 27 +++++++++++++++++++++++++++
 src/ext4.c     | 28 ++--------------------------
 2 files changed, 29 insertions(+), 26 deletions(-)

diff --git a/include/ext4.h b/include/ext4.h
index 981e517..68b33c9 100644
--- a/include/ext4.h
+++ b/include/ext4.h
@@ -51,6 +51,8 @@ extern "C" {
 #include <ext4_oflags.h>
 #include <ext4_debug.h>
 
+#include <ext4_fs.h>
+#include <ext4_journal.h>
 #include <ext4_blockdev.h>
 
 /********************************OS LOCK INFERFACE***************************/
@@ -65,6 +67,31 @@ struct ext4_lock {
     void (*unlock)(void);
 };
 
+/**@brief   Mount point descriptor.*/
+struct ext4_mountpoint {
+
+    /**@brief   Mount done flag.*/
+    bool mounted;
+
+    /**@brief   Mount point name (@ref ext4_mount)*/
+    char name[CONFIG_EXT4_MAX_MP_NAME + 1];
+
+    /**@brief   OS dependent lock/unlock functions.*/
+    const struct ext4_lock *os_locks;
+
+    /**@brief   Ext4 filesystem internals.*/
+    struct ext4_fs fs;
+
+    /**@brief   JBD fs.*/
+    struct jbd_fs jbd_fs;
+
+    /**@brief   Journal.*/
+    struct jbd_journal jbd_journal;
+
+    /**@brief   Block cache.*/
+    struct ext4_bcache bc;
+};
+
 /********************************FILE DESCRIPTOR*****************************/
 
 /**@brief   File descriptor. */
diff --git a/src/ext4.c b/src/ext4.c
index 74458f2..be696d4 100644
--- a/src/ext4.c
+++ b/src/ext4.c
@@ -71,31 +71,6 @@
             (_m)->os_locks->unlock();                              \
     } while (0)
 
-/**@brief   Mount point descriptor.*/
-struct ext4_mountpoint {
-
-    /**@brief   Mount done flag.*/
-    bool mounted;
-
-    /**@brief   Mount point name (@ref ext4_mount)*/
-    char name[CONFIG_EXT4_MAX_MP_NAME + 1];
-
-    /**@brief   OS dependent lock/unlock functions.*/
-    const struct ext4_lock *os_locks;
-
-    /**@brief   Ext4 filesystem internals.*/
-    struct ext4_fs fs;
-
-    /**@brief   JBD fs.*/
-    struct jbd_fs jbd_fs;
-
-    /**@brief   Journal.*/
-    struct jbd_journal jbd_journal;
-
-    /**@brief   Block cache.*/
-    struct ext4_bcache bc;
-};
-
 /**@brief   Mountpoints.*/
 static struct ext4_mountpoint s_mp[CONFIG_EXT4_MOUNTPOINTS_COUNT];
 
@@ -399,7 +374,8 @@ Finish:
     return r;
 }
 
-static struct ext4_mountpoint *ext4_get_mount(const char *path)
+__attribute__((weak))
+struct ext4_mountpoint *ext4_get_mount(const char *path)
 {
     struct ext4_mountpoint *mp = NULL;
     uint32_t fspath, prefixlen;
-- 
2.39.2

